### PR TITLE
Support multiple UploadParams in upload service

### PR DESCRIPTION
**Pull request by barbelity**

### CODE REVIEW

File: artifactory/manager.go [hunk 1]
```diff
@@ -130,12 +130,11 @@ func (sm *ArtifactoryServicesManager) DeleteProps(params services.PropsParams) (
 	return setPropsService.DeleteProps(params)
 }
 
-func (sm *ArtifactoryServicesManager) UploadFiles(params services.UploadParams) (artifactsFileInfo []utils.FileInfo, totalUploaded, totalFailed int, err error) {
+func (sm *ArtifactoryServicesManager) UploadFiles(params ...services.UploadParams) (artifactsFileInfo []utils.FileInfo, totalUploaded, totalFailed int, err error) {
```
Comment (eyalbe4) [inline artifactory/manager.go]: The README should be updated with the new API signature.

File: artifactory/services/upload.go [hunk 2]
```diff
@@ -26,7 +26,6 @@ type UploadService struct {
 	DryRun            bool
 	Threads           int
 	MinChecksumDeploy int64
-	Retries           int
```
Comment (eyalbe4) [inline artifactory/services/upload.go]: The *Retries* option has been removed from the upload service. 
This causes a bit of inconsistency with the download service, which still includes the *Retries* option.
I think that it make more sense for the *Retries* to be set in the service level than in the upload / download method level, but we can discuss this further.

File: artifactory/manager.go (hunk 1 above)
Comment (barbelity) [reply artifactory/manager.go]: The README is not affected by this change.

File: artifactory/services/upload.go (hunk 2 above)
Comment (barbelity) [reply artifactory/services/upload.go]: This will be fixed in a future development which redesigns the way of working with services and params.

**Pull request merged**